### PR TITLE
DOC: use macros

### PR DESCRIPTION
**Pull request by sinhrks**
Use ``matrix`` and ``vector`` macros in API docs.

### CODE REVIEW

File: src/matrix/decomposition/hessenberg.rs [hunk 1]
```diff
@@ -85,15 +90,19 @@ impl<T: Any + Float> Matrix<T> {
...
     ///
-    /// let a = Matrix::new(3,3,vec![1.,2.,3.,4.,5.,6.,7.,8.,9.]);
+    /// let a = matrix![1., 2., 3.;
+    ///                 4., 5., 6.;
+    ///                 7., 8., 9.];
     ///
     /// // u is the transform, h is the upper hessenberg form.
     /// let (u,h) = a.clone().upper_hess_decomp().expect("This matrix should decompose!");
     ///
     /// println!("The hess : {:?}", h.data());
```
Comment (AtheMathmo) [inline src/matrix/decomposition/hessenberg.rs]: These are also a little weird (my fault). Could we change this to:

```rust
println!("The hessian: {}", h);
println!("Transforming a to hessian: {}", u.transpose() * a * u);
```

File: src/vector/impl_vec.rs [hunk 2]
```diff
@@ -294,13 +304,15 @@ impl<T: Copy + Zero + Mul<T, Output = T> + Add<T, Output = T>> Vector<T> {
     /// # Examples
     ///
     /// ```
+    /// # #[macro_use] extern crate rulinalg; fn main() {
     /// use rulinalg::vector::Vector;
     ///
-    /// let a = Vector::new(vec![1.0,2.0,3.0,4.0]);
-    /// let b = Vector::new(vec![2.0; 4]);
+    /// let a = vector![1.0,2.0,3.0,4.0];
```
Comment (AtheMathmo) [inline src/vector/impl_vec.rs]: In the interest of consistency - we should add spacing between elements here.

File: src/vector/impl_vec.rs [hunk 3]
```diff
@@ -315,12 +327,14 @@ impl<T: Copy + Zero + Add<T, Output = T>> Vector<T> {
     /// # Examples
     ///
     /// ```
+    /// # #[macro_use] extern crate rulinalg; fn main() {
     /// use rulinalg::vector::Vector;
     ///
-    /// let a = Vector::new(vec![1.0,2.0,3.0,4.0]);
+    /// let a = vector![1.0,2.0,3.0,4.0];
```
Comment (AtheMathmo) [inline src/vector/impl_vec.rs]: And spacing here.

File: src/vector/impl_vec.rs [hunk 4]
```diff
@@ -353,13 +369,15 @@ impl<T: Copy + Div<T, Output = T>> Vector<T> {
     /// # Examples
     ///
     /// ```
+    /// # #[macro_use] extern crate rulinalg; fn main() {
     /// use rulinalg::vector::Vector;
     ///
-    /// let a = Vector::new(vec![1.0,2.0,3.0,4.0]);
-    /// let b = Vector::new(vec![1.0,2.0,3.0,4.0]);
+    /// let a = vector![1.0,2.0,3.0,4.0];
```
Comment (AtheMathmo) [inline src/vector/impl_vec.rs]: A couple more needing spaces here.

**Comment by sinhrks** [comment]
Thx, updated based on comments.

**Comment by AtheMathmo** [comment]
Thanks!

**Pull request merged**